File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.java
class Solution {
    public int maxProfit(int[] prices) {
        var max = 0;
                    for (int i =0;i<prices.length-1;i++){
                                    for(int j=i+1; j < prices.length; j++){
                                                        max = Integer.max(max, prices[j] - prices[i]);
                                                                        }
                                                                                    }
                                                                                                return max;
    }
}

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-3.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        stack = []
        largest = float("-inf")

        for index, height in enumerate(heights):
            toAppend = index
            while len(stack) > 0 and height < stack[-1][1]:
                poppedIndex, poppedHeight = stack.pop()
                toAppend = poppedIndex
                largest = max(largest, poppedHeight * (index - poppedIndex))
            stack.append((toAppend, height))

        while len(stack) > 0:
            index, height = stack.pop()
            largest = max(largest, height * (len(heights) - index))
        
        return largest
